### ISSUE TITLE
Virustotal use case is broken

### ISSUE THREAD

**Issue by vikman90** [wazuh/wazuh#9985]
|Version|Component|Installation type|Cause|
|---|---|---|---|
|4.2.0|Analysisd|Manager|#8828|

@jctello commented:

After upgrading to 4.2 the VirusTotal integration can break a long time after the Wazuh manager has been running resulting in the following errors:
```
2021/09/06 10:16:26 wazuh-integratord: ERROR: Unable to run integration for virustotal -> integrations
2021/09/06 10:16:26 wazuh-integratord: ERROR: While running virustotal -> integrations. Output: ConnectionRefusedError: [Errno 111] Connection refused
```

## Virustotal integration backtrace

```
Traceback (most recent call last):
  File "/var/ossec/integrations/virustotal.py", line 208, in <module>
    main(sys.argv)
  File "/var/ossec/integrations/virustotal.py", line 64, in main
    msg = request_virustotal_info(json_alert,apikey)
  File "/var/ossec/integrations/virustotal.py", line 139, in request_virustotal_info
    data = query_api(alert["syscheck"]["md5_after"], apikey)
  File "/var/ossec/integrations/virustotal.py", line 115, in query_api
    send_event(alert_output)
  File "/var/ossec/integrations/virustotal.py", line 183, in send_event
    sock.connect(socket_addr)
ConnectionRefusedError: [Errno 111] Connection refused
```

## Rationale

The Virustotal integration script is unable to connect to Analysisd, although it's running and working normally.

We tested a Python application to connect to Analysisd:
```python
import socket
socket.socket(socket.AF_UNIX, socket.SOCK_DGRAM).connect('/var/ossec/queue/sockets/queue')
```
And it raised the same exception:
```
ConnectionRefusedError: [Errno 111] Connection refused
```

The first issue that we detected is that the inode of the socket that Analysisd holds is different from the socket file located in the directory `sockets`:

```sh
# lsof -p `pidof wazuh-analysisd` | grep sockets/queue
wazuh-ana 1244 ossec    4u  unix 0xffff8880e4005c00       0t0 18704938 queue/sockets/queue

# ls -i /var/ossec/queue/sockets/queue
16891790 /var/ossec/queue/sockets/queue
```

On the other hand, we found that the socket file time changes everytime we run a configuration test for Analysisd:
```sh
# ls -l /var/ossec/queue/sockets/queue
srw-rw---- 1 ossec ossec 0 Sep  6 17:36 /var/ossec/queue/sockets/queue

# /var/ossec/bin/wazuh-analysisd -t

# ls -l /var/ossec/queue/sockets/queue
srw-rw---- 1 ossec ossec 0 Sep  6 17:41 /var/ossec/queue/sockets/queue
```

In conclusion, Analysisd refreshes the socket file on configuration testing. That prevents components from connecting from that moment on. However, components that are already connected won't be affected.

## Proposed fix

1. Move the queue startup code to the previous place.
2. Prevent Analysisd from creating a PIDfile if it's running in testing mode.
